File: source/algorithm/NQ039/NQ039_improved.cpp
// NQ039 合法标识符
#include <cctype>
#include <iostream>
#include <string>
using namespace std;

int main() {
    int n;
    cin >> n;
    while (n--) {
        string s;
        cin >> s;
        if (isdigit(static_cast<unsigned char>(s[0]))) {
            cout << "no" << endl;
            continue;
        }
        bool valid = true;
        for (char c : s) {
            unsigned char uc = static_cast<unsigned char>(c);
            if (!(isdigit(uc) || isalpha(uc) || c == '_')) {
                valid = false;
                break;
            }
        }
        cout << (valid ? "yes" : "no") << endl;
    }
    return 0;
}

File: 兴趣班入门百练讲义/NQ100/NQ039/NQ039_improved.py
# NQ039 合法标识符
def main():
    import sys
    input_data = [line.strip() for line in sys.stdin]
    ptr = 0
    n = int(input_data[ptr])
    ptr += 1
    for _ in range(n):
        if ptr >= len(input_data):
            break
        s = input_data[ptr]
        ptr += 1
        if s[0].isdigit():
            print("no")
            continue
        valid = True
        for c in s:
            if not (c.isdigit() or c.isalpha() or c == '_'):
                valid = False
                break
        print("yes" if valid else "no")

if __name__ == "__main__":
    main()